### PR TITLE
refactor:  move transak plugin

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugins/Transak/src/hooks/useTransakAllowanceCoin.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
+import type { Coin } from '@masknet/web3-providers'
```
Comment (guanbinrui) [inline packages/plugins/Transak/src/hooks/useTransakAllowanceCoin.ts]: Refactor to use `FungibleTokenDetailed` or `ERC20TokenDetailed`.

File: packages/dashboard/src/API.tsx [hunk 2]
```diff
@@ -7,6 +7,7 @@ import type { Services as ServiceType } from '../../mask/dist/src/extension/serv
 import type { MaskEvents } from '@masknet/shared-base'
 import type { WebExtensionMessage } from '@dimensiondev/holoflows-kit'
 import type { WalletMessages } from '@masknet/plugin-wallet'
+import type { PluginTransakMessages } from '@masknet/plugin-transak'
```
Comment (Jack-Works) [inline packages/dashboard/src/API.tsx:10]:
```suggestion
import type { TransakMessages } from '@masknet/plugin-transak'
```

Comment (UncleBill) [reply packages/dashboard/src/API.tsx:10]: I think we should not add the message manager of an individual plugin to PluginMessages anymore. Let's import it from its package and use it directly.

```diff
diff --git a/packages/dashboard/src/components/PageFrame/FeaturePromotions/index.tsx b/packages/dashboard/src/components/PageFrame/FeaturePromotions/index.tsx
index b9db9eee3..ae45d0afd 100644
--- a/packages/dashboard/src/components/PageFrame/FeaturePromotions/index.tsx
+++ b/packages/dashboard/src/components/PageFrame/FeaturePromotions/index.tsx
@@ -3,7 +3,8 @@ import { useNavigate } from 'react-router-dom'
 import { makeStyles } from '@masknet/theme'
 import { useRemoteControlledDialog } from '@masknet/shared'
 import { useAccount } from '@masknet/web3-shared-evm'
-import { PluginMessages, Services } from '../../../API'
+import type { TransakMessages } from '@masknet/plugin-transak'
+import { Services } from '../../../API'
 import { PersonaContext } from '../../../pages/Personas/hooks/usePersonaContext'
 import { DashboardRoutes } from '@masknet/shared-base'
 import { PluginId } from '@masknet/plugin-infra'
@@ -37,7 +38,7 @@ export const FeaturePromotions = memo(() => {
     const account = useAccount()

     const { currentPersona, connectPersona } = PersonaContext.useContainer()
-    const { setDialog: setBuyDialog } = useRemoteControlledDialog(PluginMessages.Transak.buyTokenDialogUpdated)
+    const { setDialog: setBuyDialog } = useRemoteControlledDialog(TransakMessages.buyTokenDialogUpdated)

     const isConnectedTwitter = useMemo(() => {
         if (!currentPersona) return false

```

File: .vscode/settings.json [hunk 3]
```diff
@@ -53,5 +53,24 @@
         "editor.defaultFormatter": "esbenp.prettier-vscode",
         "editor.wordWrap": "on",
         "editor.quickSuggestions": true
-    }
+    },
+    "i18n-ally.localesPaths": [
+        "packages/dashboard/src/locales",
```
Comment (Jack-Works) [inline .vscode/settings.json]: Please move this to another PR and integrate this with `npx gulp new-pkg` script.

File: packages/dashboard/src/API.tsx [hunk 4]
```diff
@@ -18,7 +18,6 @@ export interface PluginServices extends DashboardPluginServices {
 }
 export interface PluginMessages extends DashboardPluginMessages {
     Wallet: typeof WalletMessages
-    Transak: typeof import('../../mask/dist/src/plugins/Transak/messages').PluginTransakMessages
```
Comment (Jack-Works) [inline packages/dashboard/src/API.tsx:21]: Please do not access `TransakMessages`, but use ways like how we handle `Wallet`.

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/API.tsx (hunk 2 above)
Comment (Jack-Works) [reply packages/dashboard/src/API.tsx:10]: the dashboard is intended to be an isolated project (although the intention is failing, we should not make it worse). it should not directly access other packages in this way.

**Comment by Jack-Works** [comment]
Too many merge conflicts. I suggest starting a new PR.
